tools/snapshot/node_mksnapshot.cc

```diff
@@ -0,0 +1,53 @@
+#include <cstdio>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include "libplatform/libplatform.h"
+#include "node_internals.h"
+#include "snapshot_builder.h"
+#include "v8.h"
+
+#ifdef _WIN32
+#include <VersionHelpers.h>
+#include <WinError.h>
+#include <windows.h>
+
+int wmain(int argc, wchar_t* argv[]) {
+#else   // UNIX
+int main(int argc, char* argv[]) {
+#endif  // _WIN32
+
+  if (argc < 2) {
+    std::cerr << "Usage: " << argv[0] << " <path/to/output.cc>\n";
+    return 1;
+  }
+
+  std::ofstream out;
+  out.open(argv[1], std::ios::out | std::ios::binary);
+  if (!out.is_open()) {
+    std::cerr << "Cannot open " << argv[1] << "\n";
+    return 1;
+  }
+
+  int node_argc = 1;
+  char argv0[5] = "node";
+  char* node_argv[] = {argv0};
+
+  node::InitializationResult result =
+      node::InitializeOncePerProcess(node_argc, node_argv);
```

Comment:
`node::InitializeOncePerProcess(1, &argv0)`?

That said, it might not be a bad idea to keep the existing code but nullptr-terminate the `node_argv` array.

Reply:
`char (*)[5]` does not decay to `char **`, we could cast here but I guess we might as well just keep it. I'll add a nullptr at the end.